i3lock.c

```diff
@@ -632,6 +632,36 @@ void handle_screen_resize(void) {
     redraw_screen();
 }
 
+static bool verify_png_image(char *image_path) {
+    if (!image_path) {
+        return false;
+    }
+
+    /* Check file exists and has correct PNG header */
+    FILE *png_file = fopen(image_path, "r");
+    if (png_file == NULL) {
+        fprintf(stderr, "Image file path \"%s\" cannot be opened\n", image_path);
```

Comment:
Let’s tell the user why the file could not be opened: please include strerror(errno) in the message. Please also verify that the error message is printed as you’d expect when specifying a file that doesn’t access (or where you don’t have read permission).

Reply:
Just added that. For the 2 tests you mention, here are the messages:
- Image file path "foo" cannot be opened: "No such file or directory"
- Image file path "bar" cannot be opened: "Permission denied"